### PR TITLE
`Development`: Replace learning goal with competency in comments

### PR DESCRIPTION
**Pull request by terlan98**
#### General
- [ ] ~~This is a small issue that I tested locally and was confirmed by another developer on a test server.~~
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).

#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I documented the TypeScript code using JSDoc style.

### Description

This PR is a part of follow-up PRs on #6388. There's currently an inconsistency between the usage of 'learning goal' and 'competency' throughout the system. This PR **only addresses the comments**, replacing all occurrences of the term 'learning goal' with 'competency'.

### Steps for Testing
1. Take a look at the changes and verify that the replacements are correct

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
- [x] Code Review 3
- [x] Code Review 4

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/domain/LearningGoalRelation.java [hunk 1]
```diff
@@ -3,7 +3,7 @@
 import javax.persistence.*;
 
 /**
- * This class models the relation between two learning goal. Imagine a graph: (tail) --- type --> (head)
+ * This class models the relation between two competency. Imagine a graph: (tail) --- type --> (head)
```
Comment (laadvo) [inline src/main/java/de/tum/in/www1/artemis/domain/LearningGoalRelation.java:6]:
```suggestion
 * This class models the relation between two competencies. Imagine a graph: (tail) --- type --> (head)
```

**Review by laadvo — COMMENTED** [review]
I just found one small instance that should be fixed , otherwise the comments seem to be okay. I was not able to find any other comments that still include "learning goal" that you might have missed.

**Review by laadvo — APPROVED** [review]
The changes look good :+1:

**Review by tobias-lippert — APPROVED** [review]
Replacements look good to me, couldn't find any missed occurrences

**Review by bassner — APPROVED** [review]

**Review by SolizerCodes — APPROVED** [review]

**Pull request merged**